components/timestamp/index.test.tsx

```diff
@@ -37,14 +37,18 @@ describe('mapStateToProps', () => {
         },
     } as unknown as GlobalState;
 
+    function deepCopy(object: any) {
+        return JSON.parse(JSON.stringify(object));
+    }
+
     describe('hourCycle', () => {
         test('hourCycle should be h12 when military time is false and the prop was not set', () => {
             const props = mapStateToProps(initialState, {});
             expect(props.hourCycle).toBe('h12');
         });
 
         test('hourCycle should be h23 when military time is true and the prop was not set', () => {
-            const testState = {...initialState};
```

Comment:
Spread operator {...} make a shallow copy of nested object. So they were replaced in unit tests for timestamp. This was making createSelector() who relies on reference comparisons to determine if inputs have changed to think that nothing was change because value was updated in the original object and new object from the shallow copy.
I created a true deep copy by using `JSON.parse(JSON.stringify(object))`.